File: container_server/src/cmmp/LoginPacket.cpp
#include "cmmp/LoginPacket.hpp"

namespace cmmp {

const PacketId LoginPacket::id = PacketId::Login;

LoginPacket LoginPacket::decode(std::istream& is) {
    std::string username = StreamUtils::read<std::string>(is);
    std::string password = StreamUtils::read<std::string>(is);
    bool newUser = StreamUtils::read<bool>(is);

    return LoginPacket(username, password, newUser);
}

void LoginPacket::encode(std::ostream& os) const {
    StreamUtils::write(os, this->username);
    StreamUtils::write(os, this->password);
    StreamUtils::write(os, this->newUser);
}

}

File: iobrep/src/main/java/be/hepl/benbear/iobrep/LoginPacket.java
package be.hepl.benbear.iobrep;

public class LoginPacket extends Packet {

    private final String username;
    private final String password;

    public LoginPacket(String username, String password) {
        super(PacketId.LOGIN);
        this.username = username;
        this.password = password;
    }

    public String getUsername() {
        return username;
    }

    public String getPassword() {
        return password;
    }

}
